### PR TITLE
[BugFix] Fix table is lost bug where create/drop db and table concurrently

### PR DESCRIPTION
**Pull request by gengjun-git**
Fixes SR-19348
The following scenes will lose table:
T1: begin to create t (id=1) on db d(id=2)
T2: db d(id=2) is dropped
T3: db d(id=3) is created
T4: table t(id=1) is created to db d(id=2)
T5: table t(id=5) is created to db d(id=3)
T6: there are some data load on t(id=5)
When system restarts, there will be two create table log, but only the first will success, which is table t(id=1), but all the operations is based on t(id=5).

The key point is on time T4, the check of database existence is based on db name, if we check based on id, the creation of t(id=1) will fail.

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by sduzh** [comment]
Looks like it's the same issue as #28940

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 4 / 7 (57.14%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/transaction/DatabaseTransactionMgr.java|1|4|25.00%|[1527, 1528, 1529]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|3|3|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/transaction/DatabaseTransactionMgr.java [hunk 1]
```diff
@@ -1523,6 +1523,11 @@ private void updateCatalogAfterCommitted(TransactionState transactionState, Data
         for (TableCommitInfo tableCommitInfo : transactionState.getIdToTableCommitInfos().values()) {
             long tableId = tableCommitInfo.getTableId();
             Table table = db.getTable(tableId);
+            if (table == null) {
```
Comment (sduzh) [inline fe/fe-core/src/main/java/com/starrocks/transaction/DatabaseTransactionMgr.java:1526]: If the table was force dropped, isn't it possible to have `table==null` here as well?

File: fe/fe-core/src/main/java/com/starrocks/transaction/DatabaseTransactionMgr.java [hunk 2]
```diff
@@ -1523,6 +1523,11 @@ private void updateCatalogAfterCommitted(TransactionState transactionState, Data
...
+                LOG.warn("updateCatalogAfterCommitted table is null, table: {}, db: {}-{}",
+                        tableId, db.getId(), db.getFullName());
+                return;
```
Comment (sduzh) [inline fe/fe-core/src/main/java/com/starrocks/transaction/DatabaseTransactionMgr.java:1529]: If this is a multi-table transaction and other tables sill exists, is it ok to return directly here? How about replace the `return` with `continue`?

File: fe/fe-core/src/main/java/com/starrocks/transaction/DatabaseTransactionMgr.java [hunk 3]
```diff
@@ -1523,6 +1523,11 @@ private void updateCatalogAfterCommitted(TransactionState transactionState, Data
         for (TableCommitInfo tableCommitInfo : transactionState.getIdToTableCommitInfos().values()) {
             long tableId = tableCommitInfo.getTableId();
             Table table = db.getTable(tableId);
+            if (table == null) {
+                LOG.warn("updateCatalogAfterCommitted table is null, table: {}, db: {}-{}",
```
Comment (sduzh) [inline fe/fe-core/src/main/java/com/starrocks/transaction/DatabaseTransactionMgr.java:1527]:
```suggestion
                LOG.warn("table is null, table: {}, db: {}-{}",
```
The log4j will print the method name itself, so there is no need to print the method name here.

Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/transaction/DatabaseTransactionMgr.java]: This is debug log, I have removed it.

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|3|3|100.00%|[]|

**Review by satanson — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**